### ISSUE TITLE
序列特征输入问题

### ISSUE THREAD

**Issue by duanxian** [PaddlePaddle/Paddle#14846]
网络输入层定义如下：
```
lr_merged_input = fluid.layers.data(name="lr_merged_input", shape=[1], dtype="float32", lod_level=1)
dnn_merged_input = fluid.layers.data(name='dnn_merged_input', shape=[26], dtype='float32', lod_level=0)
click = fluid.layers.data(name = "click", shape = [1], dtype = "int64", lod_level=0)
```
数据feed函数定义：
```
def read_train():
    with open("trans_data/transfered.train.txt") as f:
        for line_id, line in enumerate(f):
            fs = line.strip().split('\t')
            lr_input = map(int, fs[0].split())
            dnn_input = map(int, fs[1].split())
            click = map(int, fs[2].split())
            res = []
            res.append(lr_input)
            res.append(dnn_input)
            res.append(click)
            yield res
```
训练代码：
```
trainer.train(
        reader=train_reader,
        num_epochs=3,
        event_handler=event_handler,
        feed_order=['lr_merged_input', 'dnn_merged_input', 'click'])
```

输入数据样例：
[[[7, 201, 2852, 3803, 6927, 26986, 29115, 29704, 30298, 31762, 31778, 31859, 31932], [14, 794, 2333, 45954, 70097, 70247, 70538, 77885, 78138, 78730, 89172, 94178, 134284, 137051, 138867, 143474, 176934, 177478, 179494, 180799, 182006, 219422, 219443, 220177, 231791, 232390], [0]]]

报错：
paddle.fluid.core.EnforceNotMet: enforce out_dims[j] == ins[i][j] failed, 1 != 13
Input tensors should have the same elements except the specify axis. at [/paddle/paddle/fluid/operators/concat_op.cc:52]

请问是怎么回事？

**Comment by duanxian** [comment]
贴上源代码：
```
#coding:utf-8

from __future__ import print_function
import paddle as paddle
import paddle.fluid as fluid
import six
import sys

try:
    from paddle.fluid.contrib.trainer import *
    from paddle.fluid.contrib.inferencer import *
except ImportError:
    print(
        "In the fluid 1.0, the trainer and inferencer are moving to paddle.fluid.contrib",
        file=sys.stderr)
    from paddle.fluid.trainer import *
    from paddle.fluid.inferencer import *
import numpy
import numpy as np
import sys
from functools import partial

import math
import os

def inference_program(is_sparse):
    lr_merged_input = fluid.layers.data(name="lr_merged_input", shape=[1], dtype="float32", lod_level=1)
    dnn_merged_input = fluid.layers.data(name='dnn_merged_input', shape=[26], dtype='float32', lod_level=0)

    dnn_layer_dims = [128, 64, 32, 1]
    dnn_embedding = fluid.layers.fc(input = dnn_merged_input, size = dnn_layer_dims[0], act='sigmoid')
    _input_layer = dnn_embedding
    for i, dim in enumerate(dnn_layer_dims[1:]):
        fc = fluid.layers.fc(input=_input_layer, size=dim, act='relu', name='dnn-fc-%d' % i)
        _input_layer = fc

    lr_embedding = fluid.layers.fc(input = lr_merged_input, size = 1, act='relu')

    concat_embed = fluid.layers.concat(
        input=[_input_layer, lr_embedding], axis=1)
    output = fluid.layers.fc(
        input = concat_embed, size = 2, act = 'softmax')

    return output
def train_program(is_sparse):
    pctr = inference_program(is_sparse)
    click = fluid.layers.data(name = "click", shape = [1], dtype = "int64", lod_level=0)
    cost = fluid.layers.cross_entropy(input = pctr, label = click)
    avg_cost = fluid.layers.mean(cost)
    return avg_cost

def optimizer_func():
    return fluid.optimizer.AdagradOptimizer(
        learning_rate=3e-3,
        regularization=fluid.regularizer.L2DecayRegularizer(8e-4))

def read_train():
    with open("trans_data/transfered.train.txt") as f:
        for line_id, line in enumerate(f):
            fs = line.strip().split('\t')
            lr_input = map(int, fs[0].split())
            dnn_input = map(int, fs[1].split())
            click = map(int, fs[2].split())
            res = []
            res.append(lr_input)
            res.append(dnn_input)
            res.append(click)
            yield res

def read_test():
    with open("trans_data/transfered.test.txt") as f:
        for line_id, line in enumerate(f):
            fs = line.strip().split('\t')
            lr_input = np.array(map(int, fs[0].split()))
            dnn_input = np.array(map(int, fs[1].split()))
            click = np.array(map(int, fs[2].split()))
            yield lr_input, dnn_input, click

BATCH_SIZE = 1
def train(use_cuda, train_program, params_dirname):
    train_reader = paddle.batch(read_train, BATCH_SIZE)
    test_reader = paddle.batch(read_test, BATCH_SIZE)

    print(next(train_reader()))

    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()

    def event_handler(event):
        if isinstance(event, EndStepEvent):
            outs = trainer.test(
                reader=test_reader,
                feed_order=['lr_merged_input', 'dnn_merged_input', 'click'])
            avg_cost = outs[0]

            if event.step % 10 == 0:
                print("Step %d: Average Cost %f" % (event.step, avg_cost))

            if avg_cost < 0.5:
                trainer.save_params(params_dirname)
                trainer.stop()

            if math.isnan(avg_cost):
                sys.exit("got NaN loss, training failed.")

    trainer = Trainer(
        train_func=train_program, optimizer_func=optimizer_func, place=place)
    print("here5")
    trainer.train(
        reader=train_reader,
        num_epochs=3,
        event_handler=event_handler,
        feed_order=['lr_merged_input', 'dnn_merged_input', 'click'])

def main(use_cuda, is_sparse):
    if use_cuda and not fluid.core.is_compiled_with_cuda():
        return

    params_dirname = "pctr.inference.model"

    train(
        use_cuda=use_cuda,
        train_program=partial(train_program, is_sparse),
        params_dirname=params_dirname)

if __name__ == '__main__':
    main(use_cuda=False, is_sparse=True)
```

**Comment by duanxian** [comment]
使用的训练数据样例：
```
7 201 2852 3803 6927 26986 29115 29704 30298 31762 31778 31859 31932    14 794 2333 45954 70097 70247 70538 77885 78138 78730 89172 94178 134284 137051 138867 143474 1
76934 177478 179494 180799 182006 219422 219443 220177 231791 232390    0
3 464 2852 3828 12394 27050 29033 29707 29983 31763 31771 31859 31969   1 667 11221 51857 70111 70243 74353 77881 78138 80955 90730 102631 135595 137051 139863 150754 
176934 178052 179495 180800 189974 219422 219435 219641 231786 231839   0
3 158 2853 3797 12395 27018 29031 29662 29919 31763 31773 31862 31935   4 555 11222 51858 70097 70243 70321 77881 78138 78154 89157 102632 134270 137057 139207 150755 
176941 177270 179964 180799 189975 219422 219435 222191 231787 231870   0
2 284 2874 3799 7412 27149 29045 29666 29866 31762 31771 31861 31939    3 561 1136 44999 70097 70243 74744 77881 78138 78144 91098 93035 135881 137051 137117 142408 17
6940 176983 179495 180800 180898 219422 219435 219518 231786 231839     0
3 156 2851 3795 12396 26998 29036 29663 29819 31763 31774 31859 31938   1 930 8052 45443 70098 70244 73201 77881 78138 78144 91514 99634 134722 137052 139381 142893 17
6935 178214 179495 180800 187176 219425 219435 219843 231786 231839     0
```
每个测试用例有40个特征，前13个和后26个分别用空格隔开，中间用\t分割，最后一个为label，也是\t分割

**Comment by wangguibao** [comment]
调试后发现用户program的_input_layer, lr_embedding 2个variable，一个是Tensor, 一个是LoDTensor，无法用concat去连接；另外_input_layer的shape是1 x 1, lr_embedding shape是 13 x 1，在axis=1方向也不能concat
